addons/sale/models/sale.py

```diff
@@ -577,6 +577,18 @@ def _compute_qty_delivered_updateable(self):
         for line in self:
             line.qty_delivered_updateable = (line.order_id.state == 'sale') and (line.product_id.track_service == 'manual') and (line.product_id.expense_policy == 'no')
 
+    @api.depends('invoice_lines.invoice_id.state', 'invoice_lines.invoice_id.refund_invoice_id.state')
```

Comment:
Yes, but what happens if I delete an invoice_lines or invoice_lines.invoice_id ? The depends won't trigger and I might have an invalid value.